asr1k_neutron_l3/plugins/l3/service_plugins/l3_extension_adapter.py

```diff
@@ -179,6 +181,7 @@ def _check_internal_net_az_hints(self, resource, event, trigger, payload, **kwar
                 raise exc
 
 
+@resource_extend.has_resource_extenders
```

Comment:
...what I meant to say is it could live as part of `ASR1KRouterPlugin`, which is similar to Neutrons `L3RouterPlugin`. But your argument to keep it here is solid, let's keep the more neutron specific stuff in the RouterPlugin and our own implementation in the PluginBase.